### PR TITLE
Feature: Added support for closing the left pane

### LINKED ISSUE #3342: Feature: Add the ability to close the primary/left panel
**Requirements**
- Add support for closing the left pane

### PR DESCRIPTION
**Pull request by 0x5bfa**
### Resolved / Related Issues

*Internally planned.*

For the record, this implementation avoids major refactoring of removing left/right pane to be able to create panes more than two panes.

Closes #3342

### Steps used to test these changes

Stability is a top priority for Files and all changes are required to go through testing before being merged into the repo. Please include a list of steps that you used to test this PR.

1. Open Files app
2. Open right pane
3. Focus left pane
4. Close active pane from the menu

### CODE REVIEW

File: src/Files.App/Views/ShellPanesPage.xaml.cs [hunk 1]
```diff
@@ -176,23 +176,23 @@ public bool IsRightPaneVisible
 				if (value != _IsRightPaneVisible)
 				{
 					_IsRightPaneVisible = value;
```
Comment (yaira2) [inline src/Files.App/Views/ShellPanesPage.xaml.cs:178]: I suppose we still want to remove this property in a future PR.

Comment (0x5bfa) [reply src/Files.App/Views/ShellPanesPage.xaml.cs:178]: Definitely yes. But we have to do it slowly not in the same PR. I realized that in my previous trial 😅

**Comment by yaira2** [comment]
More testing is needed but I haven't found any issues on my side.

**Comment by 0x5bfa** [comment]
It's weird that when I tested on local it worked...?

Now works

**Comment by yaira2** [comment]
I'm still having this issue

**Comment by 0x5bfa** [comment]
🥹

**Comment by yaira2** [comment]
The previously mentioned issue has been resolved. But don't worry, I found another one 😂 

1. Open two panes and scroll down in the right pane
2. Closing the left pane scrolls to the top of the list

**Comment by 0x5bfa** [comment]
Single underscore 😭😭😭😭

**Comment by yaira2** [comment]
> Single underscore 😭😭😭😭

Is this related to the issue I mentioned?

**Comment by 0x5bfa** [comment]
Yes, fixed

**Comment by yaira2** [comment]
`AAAAAAAAAAAAAAA` is the right name 😂 

I just wanted to confirm.

**Review by yaira2 — APPROVED** [review]
LGTM

**Comment by yaira2** [comment]
I found a bug,
1. Open two panes
2. Resize the window to be small enough that the right pane hides
3. Make the window larger again, and notice that the right pane navigated to the home page

**Comment by hishitetsu** [comment]
> I found a bug,
> 
> 1. Open two panes
> 2. Resize the window to be small enough that the right pane hides
> 3. Make the window larger again, and notice that the right pane navigated to the home page

This is an existing issue, so it doesn't have to be fixed in this PR.

File: src/Files.App/Actions/Navigation/CloseActivePaneAction.cs [hunk 2]
```diff
@@ -20,23 +20,20 @@ public RichGlyph Glyph
...
-
-			context.PropertyChanged += Context_PropertyChanged;
+			ContentPageContext.PropertyChanged += ContentPageContext_PropertyChanged;
 		}
 
 		public Task ExecuteAsync(object? parameter = null)
 		{
-			context.ShellPage!.PaneHolder.CloseSecondaryPane();
-
+			ContentPageContext.ShellPage!.PaneHolder.CloseActivePane();
```
Comment (hishitetsu) [inline src/Files.App/Actions/Navigation/CloseActivePaneAction.cs:32]:
```suggestion
			ContentPageContext.ShellPage?.PaneHolder.CloseActivePane();
```
It should be null safe to avoid crashes, even if IsExecutable has a null check.

File: src/Files.App/Views/ShellPanesPage.xaml.cs [hunk 3]
```diff
@@ -343,14 +344,35 @@ private void RemovePane(int index = -1)
 			// Get proper position of sizer that resides with the pane that is wanted to be removed
 			var childIndex = index * 2 - 1;
 			childIndex = childIndex >= 0 ? childIndex : 0;
+			if (childIndex + 1 > RootGrid.Children.Count)
+				return;
+
+			if (childIndex is 0)
+			{
```
Comment (hishitetsu) [inline src/Files.App/Views/ShellPanesPage.xaml.cs:351]:
```suggestion
			if (childIndex >= RootGrid.Children.Count)
				return;

			if (childIndex == 0)
			{
```
It may be a matter of taste, but this is easier for me to understand.

File: src/Files.App/Views/ShellPanesPage.xaml.cs [hunk 4]
```diff
@@ -491,7 +513,7 @@ private void Pane_GotFocus(object sender, RoutedEventArgs e)
 
 		private void Pane_RightTapped(object sender, RoutedEventArgs e)
 		{
-			if (sender != ActivePane && sender is IShellPage shellPage && shellPage.SlimContentPage is not ColumnsLayoutPage)
+			if (sender != ActivePane)
```
Comment (hishitetsu) [inline src/Files.App/Views/ShellPanesPage.xaml.cs:516]: Why did you delete these conditions? It doesn't appear to be directly related to this PR.

Comment (yaira2) [reply src/Files.App/Views/ShellPanesPage.xaml.cs:516]: Are they needed?

Comment (hishitetsu) [reply src/Files.App/Views/ShellPanesPage.xaml.cs:516]: I think they are for focusing not a pane but a column when in the column layout.

Comment (0x5bfa) [reply src/Files.App/Views/ShellPanesPage.xaml.cs:516]: When it's column view, even if you focus background wouldn't changed. You can try in preview build.

**Comment by 0x5bfa** [comment]
> > I found a bug,
> > 
> > 1. Open two panes
> > 2. Resize the window to be small enough that the right pane hides
> > 3. Make the window larger again, and notice that the right pane navigated to the home page
> 
> This is an existing issue, so it doesn't have to be fixed in this PR.

My guess is here:

https://github.com/files-community/Files/blob/bfba7d4833a59d32814fa8a6e9b115548189b68b/src/Files.App/Views/ShellPanesPage.xaml.cs#L527-L533

File: src/Files.App/Views/ShellPanesPage.xaml.cs (hunk 4 above)
Comment (hishitetsu) [reply src/Files.App/Views/ShellPanesPage.xaml.cs:516]: That is a different issue, and this change causes the focus to flicker when right-clicking a column. I think this change should be reverted.
We have had similar issues before. https://github.com/files-community/Files/issues/12285

**Review by hishitetsu — APPROVED** [review]
LGTM
